src/test/java/uk/gov/register/functional/app/RegisterRule.java

```diff
@@ -52,13 +48,30 @@ public Statement apply(Statement base, Description description) {
         Statement me = new Statement() {
             @Override
             public void evaluate() throws Throwable {
-                client = clientBuilder().build("test client");
-                authenticatedTarget = client.target("http://localhost:" + appRule.getLocalPort());
-                authenticatedTarget.register(HttpAuthenticationFeature.basicBuilder().credentials("foo", "bar").build());
+                beforeTest();
                 base.evaluate();
             }
         };
-        return wholeRule.apply(me, description);
+        return appRule.apply(me, description);
+    }
+
+    /*
+     * executed before each test, but after appRule has set up the dropwizard app
+     */
+    private void beforeTest() {
+        client = clientBuilder().build("test client");
+        authenticatedTarget = client.target("http://localhost:" + appRule.getLocalPort());
+        authenticatedTarget.register(HttpAuthenticationFeature.basicBuilder().credentials("foo", "bar").build());
+
+        migrateDb();
+        wipe();
+    }
+
+    private void migrateDb() {
+        RegisterConfiguration configuration = appRule.getConfiguration();
+        Flyway flyway = configuration.getFlywayFactory().build(configuration.getDatabase().build(appRule.getEnvironment().metrics(), "flyway_db"));
+        flyway.setBaselineVersionAsString("0");
```

Comment:
Can we remove this? It doesn't appear to be set anywhere outside of the app e.g. from run-application.sh.